slightly split existing search components

alxnddr opened the pull request:
### Description
Extract components from search components bulky files as Kyle did in https://github.com/metabase/metabase/pull/16324

I started it from scratch due to multiple features got merged in the search and it felt more convenient and safe to redo the work instead of doing a rebase of Kyle's branch.

### How to test
No visual changes expected, search should work the same way as it worked before

alxnddr committed:
slightly split existing search components

gusaiani commented on frontend/src/metabase/search/components/InfoText.jsx:
```diff
@@ -0,0 +1,87 @@
+                  loadingAndErrorWrapper={false}
+                >
+                  {({ list }) =>
+                    list && list.length > 1 ? (
```
```suggestion
                    list?.length > 1 ? (
```

gusaiani commented on frontend/src/metabase/search/components/InfoText.jsx:
```diff
@@ -0,0 +1,87 @@
+      return t`Database`;
+    case "table":
+      return (
+        <span>
```
This entire block could be extracted as a component, making the switch statement faster to understand.

alxnddr replied:
I'll address it in the next PR to avoid conflicts

gusaiani commented on frontend/src/metabase/search/components/InfoText.jsx:
```diff
@@ -0,0 +1,87 @@
+    case "segment":
+    case "metric":
+      return (
+        <span>
```
This entire block could be extracted as a component, making the switch statement faster to understand.

alxnddr replied:
I have the next PR that contains extracting https://github.com/metabase/metabase/pull/17582

gusaiani commented on frontend/src/metabase/search/components/SearchResult.styled.jsx:
```diff
@@ -0,0 +1,97 @@
+  display: block;
+  background-color: transparent;
+  min-height: ${props => (props.compact ? "36px" : "54px")};
+  padding-top: 8px;
```
Could use `space(1)`

gusaiani commented on frontend/src/metabase/search/components/SearchResult.styled.jsx:
```diff
@@ -0,0 +1,97 @@
+  background-color: transparent;
+  min-height: ${props => (props.compact ? "36px" : "54px")};
+  padding-top: 8px;
+  padding-bottom: 8px;
```
Could use `space(1)`

gusaiani commented on frontend/src/metabase/search/components/SearchResult.styled.jsx:
```diff
@@ -0,0 +1,97 @@
+  padding-top: 8px;
+  padding-bottom: 8px;
+  padding-left: 14px;
+  padding-right: ${props => (props.compact ? "20px" : "32px")};
```
32px could be `space(3)`

gusaiani commented on frontend/src/metabase/search/components/SearchResult.styled.jsx:
```diff
@@ -0,0 +1,97 @@
+`;
+
+export const Description = styled(Text)`
+  padding-left: 8px;
```
could be `space(1)`

gusaiani commented on frontend/src/metabase/search/components/SearchResult.styled.jsx:
```diff
@@ -0,0 +1,97 @@
+
+export const Description = styled(Text)`
+  padding-left: 8px;
+  margin-top: 6px !important;
```
maybe could also be `space(1)`, would change from 6 to 8 px though

gusaiani commented on frontend/src/metabase/nav/components/SearchResults.jsx:
```diff
@@ -0,0 +1,48 @@
+                </li>
+              ))
+            ) : (
+              <li className="flex flex-column align-center justify-center p4 text-medium text-centered">
```
Maybe consider converting from className styles to a styled component.

alxnddr replied:
Used EmptyState here

gusaiani reviewed (commented):
👍

Some inlined comments.

How about taking the time to add unit tests, for example to the newly extracted components?

alxnddr committed:
address review comments

alxnddr committed:
review fixes

alxnddr commented:
@gusaiani thanks for the review, I addressed your comments. I think InfoText component would be nice to test, I have some changes in it in the next PR https://github.com/metabase/metabase/pull/17582 so I'll add tests there

gusaiani reviewed (approved):
Awesome Sasha, onwards o/
